streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -714,7 +720,10 @@ public StreamThread(final Time time,
     @Override
     public void run() {
         log.info("Starting");
-        setState(State.RUNNING);
+        if (!setState(State.RUNNING)) {
```

Comment:
Could this ever happen? I.e. instead of checking the lower-level tread state here, we already check the higher-level instance state that in `KafkaStreams#start` if the instance state is not `CREATED` we will fail and throw?

Reply:
If the client does:
```
KafkaStreams ks = ...
ks.start();
ks.close();
```
Isn't it possible that the StreamThread.shutdown() gets called, sees that the thread is still in CREATED state and starts shutting the stream thread down before the thread itself gets scheduled or gets far enough into run() ?

Reply:
Ack, makes sense.